### PR TITLE
Clean up Partitioned Epoch Rewards feature logic

### LINKED ISSUE #426: Feature Gate: Enable partitioned epoch rewards
### SIMD

https://github.com/solana-foundation/solana-improvement-documents/blob/main/proposals/0118-partitioned-epoch-reward-distribution.md

### Description

The [old feature gate](https://github.com/solana-labs/solana/issues/32166) wrapped an implementation that was superseded by a newer SIMD.

Partitioned epoch rewards is a design to distribute epoch stake rewards across multiple blocks to address the current performance problems associated with epoch reward distribution in a first block of a new epoch.

### Feature ID

(9bn2vTJUsUcnpiZWbu2woSKtTGW3ErZC9ERv88SDqQjK)
PERzQrt5gBD1XEe2c9XdFWqwgHY3mr7cYWbm5V772V8

### Activation Method

Single Core Contributor

### Deployment Considerations

Needs testing on a test cluster with sufficient stake accounts to trigger multiple partitions. Tests should also include rebooting from a snapshot during calculation and distribution.
Distribution should be monitored for correctness and performance.

### Minimum Beta Version

(v2.0.0)
v2.0.9

### Minimum Stable Version

(v2.0.0)
v2.0.9

### Testnet Activation Epoch

(679)

### Devnet Activation Epoch

790

### Mainnet-Beta Activation Epoch

707

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
The feature gate `partitioned_epoch_rewards_superfeature` has been activated on all public clusters, but the gating logic remains.

#### Summary of Changes
- Remove feature gate logic, primarily in `bank.rs`, but also the superfeature fix in `bank::partitioned_epoch_rewards::sysvar::create_epoch_rewards_sysvar()`
- Fixup some tests to expect PER
- Remove pre-PER epoch-boundary rewards code, now unused
- Cherry-pick @HaoranYi 's commit to remove partitioned reward test code

Fixes #426

### CODE REVIEW

File: runtime/src/bank/tests.rs [hunk 1]
```diff
@@ -1899,10 +1906,107 @@ fn test_bank_update_vote_stake_rewards() {
     });
 }
 
+impl Bank {
+    #[cfg(test)]
```
Comment (brooksprumo) [inline runtime/src/bank/tests.rs]: This file already has a top-level `#![cfg(test)]`, is this one also needed?

```suggestion
```

File: runtime/src/bank/tests.rs
Comment (brooksprumo) [inline runtime/src/bank/tests.rs:1]: Is the actual logic changing in these tests? If yes, wdyt about making those changes in a separate PR? Or are the changes coupled with removing the featurization logic?

File: runtime/src/bank.rs [hunk 2]
```diff
@@ -2420,93 +2386,6 @@ impl Bank {
...
-
-        let active_stake = if let Some(stake_history_entry) =
-            self.stakes_cache.stakes().history().get(prev_epoch)
-        {
-            stake_history_entry.effective
-        } else {
-            0
-        };
-
-        datapoint_warn!(
```
Comment (HaoranYi) [inline runtime/src/bank.rs:2494]: A question out of the scope of this PR - Are those stats useful to us? Should we port them to partitioned reward code path?

**Review by HaoranYi — DISMISSED** [review]
Nice cleanup. One of my favorite type of PRs - deleting code!
I have just one question to consider, probably to address in a follow up PR.

**Review by pgarg66 — DISMISSED** [review]
Approving from SVM

File: runtime/src/bank/tests.rs (hunk 1 above)
Comment (CriesofCarrots) [reply runtime/src/bank/tests.rs]: Nope, superfluous c+p thing. Thanks!

File: runtime/src/bank/tests.rs
Comment (CriesofCarrots) [reply runtime/src/bank/tests.rs:1]: The logic is changing slightly, and unfortunately it needs to be coupled with removing the featurization logic. The way the logic is changing is that the tests are expecting only vote rewards in the 1st block in the epoch and stake rewards in the 2nd block in the epoch, instead all rewards together in the 1st block.

We have a great many tests like this that are written with features off. So when the gating logic is removed to change the default behavior, the tests have to be updated 😕 Some of the genesis-utils functions turn all features on, which I think is probably better, even though test-writers have to juggle manually turning features off to test pre-vs-post-activation behavior.

File: runtime/src/bank.rs (hunk 2 above)
Comment (CriesofCarrots) [reply runtime/src/bank.rs:2494]: Good news! They exist in the PER code path as well (I think thanks to you) :) https://github.com/anza-xyz/agave/blob/3deac722ded065f47b36936d27480319f55dc029/runtime/src/bank/partitioned_epoch_rewards/calculation.rs#L163-L176

**Review by HaoranYi — APPROVED** [review]

**Review by pgarg66 — APPROVED** [review]

File: runtime/src/bank/tests.rs
Comment (brooksprumo) [reply runtime/src/bank/tests.rs:1]: Understood, thanks for the explanation!

**Review by brooksprumo — APPROVED** [review]
:shipit:

**Comment by CriesofCarrots** [comment]
@topointon-jump , I see that you consulted on a similar feature-gate cleanup in November. Any FD concerns with this one? Presumably you're already supporting the syscall; the only change here is to remove the feature-gating of the fn registration.

**Pull request merged**